### PR TITLE
Re-enable binary.wast spec test

### PR DESCRIPTION
**Pull request by tlively**
Fix the wast parser to accept IDs on quoted modules, remove tests that are
invalidated by the multimemory proposal, and add validation that the total
number of variables in a function is less than 2^32 and that the code section is
present if there is a non-empty function section.

### CODE REVIEW

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
PTAL at the latest commit, which imposes a much smaller limit on the number of locals to avoid allocation failure on CI.

File: src/wasm/wasm-ir-builder.cpp [hunk 1]
```diff
@@ -458,18 +461,20 @@ struct IRBuilder::ChildPopper
...
+        // The next item on the stack is the unreachable instruction we must
+        // not pop past. We cannot insert unreachables in front of it because
+        // it might be a branch we actually have to execute, so this next item
+        // must be child 0. But we are not ready to pop child 0 yet, so
+        // synthesize an unreachable instead of popping. The deeper
+        // instructions that would otherwise have been popped will remain on
+        // the stack to become prior children of future expressions or to be
+        // implicitly dropped at the end of the scope.
+        *children[i].childp = builder.builder.makeUnreachable();
+        continue;
```
Comment (kripken) [inline src/wasm/wasm-ir-builder.cpp:474]: (is this from the other PR?)

**Review by kripken — APPROVED** [review]

File: src/wasm/wasm-ir-builder.cpp (hunk 1 above)
Comment (tlively) [reply src/wasm/wasm-ir-builder.cpp]: pushed this branch so it should be updated again. That's a downside of this more manual workflow; graphite would push all the downstream branches for me automatically.

**Comment by tlively** [comment]
Now I've fixed a bunch of memory leaks by using `unique_ptr` instead of raw allocations.

**Review by kripken — APPROVED** [review]

**Pull request merged**